### PR TITLE
gas optimizations for order fills

### PR DESCRIPTION
**Pull request by nuevoalex**
A few relatively small optimizations that will save us a good bit of gas in trading

### CODE REVIEW

**Comment by nuevoalex** [comment]
This isn't necessary btw at all, but will trim off about 130k gas from binary fills and around 510k gas from cat 8 fills.

If we'd rather just not make contract changes that's ok too.

**Comment by bthaile** [comment]
I think now is the time. Ether will probably go up in price so any savings is a no-brainer.

File: source/contracts/Augur.sol [hunk 1]
```diff
@@ -103,6 +104,21 @@ contract Augur is Controlled, IAugur {
         return true;
     }
 
+    //
+    // Share Tokens
+    //
+
+    function recordMarketShareTokens(IMarket _market) public view returns (bool) {
```
Comment (epheph) [inline source/contracts/Augur.sol]: Drop view?

**Comment by epheph** [comment]
Excellent, reducing gas in trading is worthwhile.

File: source/contracts/Augur.sol (hunk 1 above)
Comment (nuevoalex) [reply source/contracts/Augur.sol]: bah! Thanks!

**Pull request merged**